controllers/csm_controller.go

```diff
@@ -1524,12 +1529,11 @@ func (r *ContainerStorageModuleReconciler) GetK8sClient() kubernetes.Interface {
 	return r.K8sClient
 }
 
-func (r *ContainerStorageModuleReconciler) GetMatchingNodes(ctx context.Context, labelKey string, labelValue string) (*corev1.NodeList, error) {
-	nodeList := &corev1.NodeList{}
-	opts := []client.ListOption{
-		client.MatchingLabels{labelKey: labelValue},
+func (r *ContainerStorageModuleReconciler) ZoneValidation(ctx context.Context, cr *csmv1.ContainerStorageModule) error {
```

Comment:
do we want to check if driver is powerflex here and skip calling the validateZones method if it isn't?
We could check
```
cr.Spec.Driver.CSIDriverType
```
for either vxflexos or powerflex

https://github.com/dell/csm-operator/blob/a892d43648ebec798e33b4968f891f06076e1990/api/v1/types.go#L78-L81

Reply:
only the powerflex precheck code will call into here.

Reply:
ah I see that now, thanks